pkg/clusters/cluster.go

```diff
@@ -46,4 +46,7 @@ type Cluster interface {
 
 	// DeleteAddon removes an existing cluster Addon.
 	DeleteAddon(ctx context.Context, addon Addon) error
+
+	// GetNodeAddresses returns a list of cluster worker node addresses
+	GetNodeAddresses(ctx context.Context) ([]string, error)
```

Comment:
If this doesn't work on GKE then I don't feel we should be adding it to the cluster interface, rather we should add it to the kind cluster type and later we can promote it into the interface if we have a general cluster implementation.

Reply:
It does work and is generally applicable to all clusters--everything does have worker nodes and those worker nodes have addresses--but GKE clusters require additional steps before the most obvious use case for this becomes viable: https://cloud.google.com/kubernetes-engine/docs/how-to/exposing-apps#creating_a_service_of_type_nodeport

I didn't try to implement that, however, as I had no need to do NodePort GKE testing, but ultimately if I did enabling access to the NodePorts would be independent of retrieving the node addresses.

Reply:
I gotcha. I think that we should add the configuration now if that seems feasible, if not let's at least make an issue. I'll consider this resolved as it doesn't need to block.